corehq/apps/reports/analytics/esaccessors.py

```diff
@@ -614,15 +615,35 @@ def _forms_with_attachments(domain, app_id, xmlns, datespan, user_types):
             pass
 
 
+def _get_forms_with_attachments(es_query):
```

Comment:
deprecates _forms_with_attachments above.
The change being the use of es query passed that already has all filters above already applied, so just the source call is added now.

Reply:
This method of deprecation means that the git history of this code will be harder to follow because you're renaming and refactoring the method in multiple steps with a period of time during which the code is duplicated. It seems better to adapt the old task to call this new one so the common code between them is preserved in this task.

However, maybe this is all a moot point given the changes I've suggested above.

Reply:
Thanks for sharing this Daniel.

This process starts with the first method being called `build_form_multimedia_zipfile`. The set of arguments change, so the need to keep the old version as well. And that trickles down to all methods being called. So if there is a task queued with old set of arguments, and the celery workers have been reloaded, they still execute.

So even if make the changes above, that still holds true. The old method wont be able to create the filters from the arguments it would receive.

let me know if I have not been able to fully understand what you were saying.

Reply:
> The old method wont be able to create the filters from the arguments it would receive.

Ah, I did not realize that. Maybe you could refactor the common logic between the two into a separate function that is called by both tasks?

Reply:
Are we still thinking of maintaining the git history?
Just thinking about the benefit of the refactor. The common logic is pretty small enough in most methods and refactor might just be an overkill, with the only exception being of `build_form_multimedia_zipfile` that has considerable overlap, so just adding new methods and removing the old ones in a follow up post rollout would just be cleaner?
Not totally against the refactor but just weighing the benefit of it.

Reply:
> Are we still thinking of maintaining the git history?

Yes, but this discussion is dragging out and probably not worth going back and forth anymore. Just do what you think is best given my feedback so far..

Reply:
Gave it a try and felt it was the right thing to do, thanks for pushing on this @millerdev 

https://github.com/dimagi/commcare-hq/pull/29552